### DISCUSSION TITLE
Dynamic controller mode for HVDC Lines

### DISCUSSION THREAD

**Discussion by AlOussama** [NREL-SIIP/PowerSimulationsDynamics.jl#173] [Q&A] [answered]
I want to ask how to include the dynamic of the inverter controller in the HVDC Line model:
As i understand, the HVDC line and the VSCDC line models just have a static control loop, but no dynamic controller (like the inverter outer and inner control loops).
That means: If i want to analyse the dynamic characterisitic of the HVDC Line, I have to build a new model using two dynamic inverters and one dc line.

**Comment by jd-lara** [comment]
Thanks for your question. At the moment we don't have a plan to include HVDC models, but we are open to discussing ways to incorporate them. Professor Milano's book has a basic VSC model that can work as a starting point.

**Reply by AlOussama** [reply]
Thanks for your reply. 
Does it mean that even the VSC constructor in powersystems is not working?
https://nrel-siip.github.io/PowerSystems.jl/stable/model_library/generated_VSCDCLine/

I will have a look in the Milano's book and than make a suggection, how to implement it, starting from the basic Milano's model.

**Reply by jd-lara** [reply, marked as answer]
The constructor works, but we haven't implemented the model in PSID yet.
